File: src/AlumniService/AlumniService/Models/Student.cs
using System;
using System.Collections.Generic;

namespace AlumniService.Models;

public partial class Student
{
    public int StudentId { get; set; }

    public int UserId { get; set; }

    public int CollegeId { get; set; }

    public virtual College College { get; set; } = null!;

    public virtual ICollection<RegisterEvent> RegisterEvents { get; set; } = new List<RegisterEvent>();

    public virtual ICollection<StudentSkillset> StudentSkillsets { get; set; } = new List<StudentSkillset>();

    public virtual User User { get; set; } = null!;
}

File: src/adminbackend/src/main/java/com/example/demo/entities/Student.java
package com.example.demo.entities;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
@Entity
@Table(name="student")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Student {
	
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int student_id;
	
	@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
	@JoinColumn(name = "user_id")
	@JsonIgnoreProperties("student")
	User user;
	
	@JsonIgnoreProperties("student")
	@ManyToOne
	 @JoinColumn(name="college_id")
	 College college;
	
	
	@OneToMany(mappedBy = "student", cascade = CascadeType.ALL, orphanRemoval = true)
	@JsonIgnoreProperties("student")
	List<RegisterEvent> registerEvents;
	
	
	 @OneToMany(mappedBy = "student", cascade = CascadeType.ALL, orphanRemoval = true)
	 @JsonIgnoreProperties("student")
	 List<StudentSkillSet> studentskillset;
	
	
	 
}
